modules/client/src/channelProvider.ts

```diff
@@ -40,33 +45,77 @@ export const createCFChannelProvider = async ({
     signerAddress: xpubToAddress(xpub),
     userPublicIdentifier: xpub,
   };
+  const connection = new RpcConnection(cfCore);
   const channelProvider = new ChannelProvider(
-    cfCore,
+    connection,
     channelProviderConfig,
     store,
     await keyGen("0"),
   );
   return channelProvider;
 };
 
-export class ChannelProvider {
-  private connection: RpcConnection;
-  private store: Store;
+export class RpcConnection extends EventEmitter implements IRpcConnection {
+  public connected: boolean = true;
+  public cfCore: CFCore;
+
+  constructor(cfCore: CFCore) {
+    super();
+    this.cfCore = cfCore;
+  }
+
+  public async send(payload: JsonRpcRequest): Promise<any> {
+    if (!this.connected) {
+      return Promise.resolve();
+    }
+    const ret = await this.cfCore.rpcRouter.dispatch({
+      id: Date.now(),
+      methodName: payload.method,
+      parameters: deBigNumberifyJson(payload.params),
+    });
+    const result = ret.result.result;
+    return result;
+  }
+
+  public on = (event: string, listener: (...args: any[]) => void): any => {
+    this.cfCore.on(event as any, listener);
+    return this.cfCore;
+  };
+
+  public once = (event: string, listener: (...args: any[]) => void): any => {
+    this.cfCore.once(event as any, listener);
+    return this.cfCore;
+  };
+
+  public open(): void {
+    this.connected = true;
+  }
```

Comment:
do we have to explicitly call `.open` when using the `RpcConnection` ? If this is all it does it seems a little unneccessary, but i get it if it's part of adhering to an accepted interface

Reply:
Yeah it's more for matching the interface, I can also make it a no-op for CF channelProvider